src/components/AppConfig/AppConfig.tsx

```diff
@@ -61,34 +53,43 @@ export const AppConfig = ({ plugin }: AppConfigProps) => {
   const [configuredSecrets, setConfiguredSecrets] = useState<SecretsSet>(initialSecrets(secureJsonFields ?? {}));
   // Whether any settings have been updated.
   const [updated, setUpdated] = useState(false);
-  const [optInUpdated, setOptInUpdated] = useState(false);
+
+  const [managedLLMOptIn, setManagedLLMOptIn] = useState<boolean>(false);
 
   const [isUpdating, setIsUpdating] = useState(false);
   const [healthCheck, setHealthCheck] = useState<HealthCheckResult | undefined>(undefined);
 
   const validateInputs = (): string | undefined => {
     // Check if Grafana-provided OpenAI enabled, that it has been opted-in
-    if (settings?.openAI?.provider === 'grafana' && !settings?.llmGateway?.isOptIn) {
+    if (settings?.openAI?.provider === 'grafana' && !managedLLMOptIn) {
       return "You must click the 'Enable OpenAI access via Grafana' button to use OpenAI provided by Grafana";
     }
     return;
   };
   const errorState = validateInputs();
 
+  const updateManagedLLMOptIn = (newOptIn: boolean): void => {
+    console.log('Try to set Opt in state to', managedLLMOptIn);
+    saveLLMOptInState(newOptIn);
```

Comment:
Do we need error handling here? In case saving the state fails.

Reply:
Yeah, I just changed our fetch functions to throw the error which gives you a nice popup and will no longer update the state.